### PR TITLE
`gp ssh`

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Adds a `gp ssh` command for printing the SSH command used to access the workspace. This relies on having SSH keys setup.
## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/11415

## How to test

```bash
go run /workspace/gitpod/components/gitpod-cli ssh
```

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

**Comment by filiptronicek** [comment]
What's left to do is to resolve this issue with getting the owner token:
```
gitpod /workspace/gitpod/components/gitpod-cli (ft/gp-ssh) $ go run . ssh --access-key
failed getting token from supervisor: rpc error: code = NotFound desc = no token available
exit status 1
```

File: components/gitpod-cli/cmd/ssh.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+			client, err := serverapi.ConnectToServer(wsInfo.GitpodApi.Endpoint, serverapi.ConnectToServerOpts{
+				Token:   clientToken.Token,
+				Context: ctx,
+				Log:     serverLog,
+			})
+			if err != nil {
+				return xerrors.Errorf("failed connecting to server: %w", err)
+			}
+
+			ownerToken, err := client.GetOwnerToken(ctx, wsInfo.WorkspaceId)
```
Comment (akosyakov) [inline components/gitpod-cli/cmd/ssh.go:75]: it should not be ever possible to access owner token on the backend. It owudl be serious vulneratility.

Comment (akosyakov) [reply components/gitpod-cli/cmd/ssh.go:75]: just print ssh command and point to how configure SSH keys

Comment (filiptronicek) [reply components/gitpod-cli/cmd/ssh.go:75]: Sounds good, thanks for the pointer. I thought about it being a potential security risk, but tried anyway. I'll remove the flag and add a bit more info to the help menu.

File: components/gitpod-cli/cmd/ssh.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"github.com/spf13/cobra"
+)
+
+// sshCmd represents the ssh command
+var sshCmd = &cobra.Command{
+	Use:   "ssh",
+	Short: "Show the SSH connection command for the current workspace",
+	Long: `Displays a command with which you can connect to the current workspace.
+The returned command requires SSH keys to be configured with Gitpod.
+See https://www.gitpod.io/docs/configure/user-settings/ssh for a guide on setting them up.
```
Comment (akosyakov) [inline components/gitpod-cli/cmd/ssh.go:24]: I would just point to `${gitpodHost}/user/keys`.  It already has learn more button. Gitpod host should be fetched from from supervisor info ednpoint.

File: components/gitpod-cli/cmd/ssh.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/gitpod"
+
+	"github.com/spf13/cobra"
+)
+
+var gitpodHost = os.Getenv("GITPOD_HOST")
```
Comment (filiptronicek) [inline components/gitpod-cli/cmd/ssh.go:20]: This is pretty ugly, but it's the best I figured for dynamically filling the Gitpod host in the help menu.

Comment (akosyakov) [reply components/gitpod-cli/cmd/ssh.go:19]: ok, but it can break in future

**Review by akosyakov — APPROVED** [review]
looks sensible, i did not try

File: components/gitpod-cli/cmd/ssh.go (hunk 3 above)
Comment (akosyakov) [reply components/gitpod-cli/cmd/ssh.go:19]: add a comment here that is a hack and should not be used in other places, that we should use info endpoint instead with TODO to remove

**Comment by filiptronicek** [comment]
Let's :shipit: 

/unhold

**Pull request merged**